PathPlanning/ReedsSheppPath/reeds_shepp_path_planning.py

```diff
@@ -110,114 +83,193 @@ def left_straight_left(x, y, phi):
     if t >= 0.0:
         v = mod2pi(phi - t)
         if v >= 0.0:
-            return True, t, u, v
+            return True, [t, u, v], ['L', 'S', 'L']
+
+    return False, [], []
+
+
+def left_straight_right(x, y, phi):
+    u1, t1 = polar(x + math.sin(phi), y - 1.0 - math.cos(phi))
+    u1 = u1 ** 2
+    if u1 >= 4.0:
+        u = math.sqrt(u1 - 4.0)
+        theta = math.atan2(2.0, u)
+        t = mod2pi(t1 + theta)
+        v = mod2pi(t - phi)
+
+        if t >= 0.0 and v >= 0.0:
+            return True, [t, u, v], ['L', 'S', 'R']
+
+    return False, [], []
 
-    return False, 0.0, 0.0, 0.0
 
+def left_x_right_x_left(x, y, phi):
```

Comment:
What does it means the “x” in the function name?

Reply:
x refers to the direction changes.